File: leetcode/976. Largest Perimeter Triangle/976.cpp
// https://helloacm.com/greedy-algorithm-to-find-the-largest-perimeter-triangle-by-sorting/
// https://leetcode.com/problems/largest-perimeter-triangle/
// EASY, ARRAY, TRIANGLE, SORTING, GREEDY

class Solution {
public:
    int largestPerimeter(vector<int>& A) {
        sort(begin(A), end(A));
        for (int i = A.size() - 3; i >= 0; -- i) {
            if (A[i] + A[i + 1] > A[i + 2]) {
                return A[i] + A[i + 1] + A[i + 2];
            }
        }
        return 0;
    }
};

File: leetcode/976. Largest Perimeter Triangle/976.py
# https://helloacm.com/teaching-kids-programming-largest-perimeter-triangle-sorting-greedy-algorithm/
# https://leetcode.com/problems/largest-perimeter-triangle/
# EASY, GREEDY

class Solution(object):
    def largestPerimeter(self, A):
        A.sort()
        for i in range(len(A) - 3, -1, -1):
            if A[i] + A[i+1] > A[i+2]:
                return A[i] + A[i+1] + A[i+2]
        return 0
